src/components/Carousel.js

```diff
@@ -641,12 +728,16 @@ class Carousel extends Component {
                                 ref={this.setListRef}
                                 {...swiperProps}
                                 allowMouseEvents={this.props.emulateTouch}>
-                              { this.renderItems() }
+                                { this.props.infiniteLoop && this.renderLastClone() }
```

Comment:
Instead of implementing 2 new methods (renderLastClone and renderFirstClone), couldn't you have done something like:

```
const items = this.renderItems()
const itemsClone = [...items];
const firstClone = itemsClone.shift();
const lastClone = itemsClone.pop();

...

<ul className={klass.SLIDER(true, this.state.swiping)} style={itemListStyles}>
    { this.props.infiniteLoop && lastClone }
    { items }
    { this.props.infiniteLoop && firstClone }
</ul>
```

Reply:
Modified `renderItems` a bit to get unique keys.